qiskit/primitives/base_estimator.py

```diff
@@ -227,12 +259,120 @@ def __call__(
             values = parameter_values[i].
 
         Args:
-            circuit_indices: the list of circuit indices.
-            observable_indices: the list of observable indices.
+            circuits: the list of circuit indices.
+            observables: the list of observable indices.
             parameter_values: concrete parameters to be bound.
             run_options: runtime options used for circuit execution.
 
         Returns:
             EstimatorResult: The result of the estimator.
+
+        Raises:
+            QiskitError: For mismatch of object id.
+            QiskitError: For mismatch of length of Sequence.
         """
+
+        # Support ndarray
+        if isinstance(parameter_values, np.ndarray):
+            parameter_values = parameter_values.tolist()
+
+        # Allow lift T to [T].
+        if circuits is not None and not isinstance(circuits, (np.ndarray, Sequence)):
+            circuits = [circuits]
+        if observables is not None and not isinstance(observables, (np.ndarray, Sequence)):
+            observables = [observables]
+        if parameter_values is not None and not isinstance(
+            parameter_values[0], (np.ndarray, Sequence)
+        ):
+            parameter_values = cast("Sequence[float]", parameter_values)
+            parameter_values = [parameter_values]
+
+        # Allow broadcasting
+        if (
+            circuits is None
+            and len(self._circuits) == 1
+            and observables is None
+            and len(self._observables) == 1
+            and parameter_values is not None
+        ):
+            circuits = [0] * len(parameter_values)
+            observables = [0] * len(parameter_values)
```

Comment:
Why is broadcast reverted? It has been removed in #7837.
Lev requested to remove both broadcast and optional. https://github.com/Qiskit/qiskit-terra/issues/7836#issuecomment-1085939623

Reply:
I tried to make both broadcasting and optional using custom decorator, but I thought the conclusion was that the default was better. I recognize that maybe, at that stage, we made it as simple as possible, and that finally a convenient interface is now put into, but I'm not sure. What is requirement in the end?

Reply:
I think it's a good opportunity to remove both as Lev requested because this PR changes the API. At least, I don't think it's a good to idea to revert the broadcast, which has been removed #7837.

Reply:
Right. I'll remove both once. If we get a request in the future, I will add it again.

Reply:
Done in d9d21a2